flight/Libraries/math/misc_math.c

```diff
@@ -192,8 +192,8 @@ void vector2_clip(float *vels, float limit)
 	float scale = mag / limit;
 
 	if (scale > 1) {
-		vels[0] /= mag;
-		vels[1] /= mag;
+		vels[0] /= scale;
+		vels[1] /= scale;
```

Comment:
Sounds reasonable for the FPU.

I don't know if you caught that I changed the conditional so that we eliminate any math until after the comparison fails. That seems like it should be a win no matter what.

This is probably the most efficient (before any optimization):

```
float mag = vectorn_magnitude(vels, 2);    // only horiz component

if (mag > limit) {
	float scale = limit / mag;
	vels[0] *= scale;
	vels[1] *= scale;
}
```